adafruit_requests.py

```diff
@@ -560,7 +560,7 @@ def request(
             ok = True
             try:
                 self._send_request(socket, host, method, path, headers, data, json)
-            except _SendFailed:
+            except (_SendFailed, OSError):
```

Comment:
@anecdata  I think this achieves your conclusion in https://github.com/adafruit/Adafruit_CircuitPython_Requests/issues/63, closing the failed socket and opening a new one.